File: num2.java
import java.util.Scanner;

public class BottleExchange {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("full bottles: ");
        int n = scanner.nextInt();
        
        System.out.print("empty bottles for exchange: ");
        int m = scanner.nextInt();
        
        int mc = n;
        int mcount = n;
        int count = 0;
        int ost = 0;
        int drinkedBottles = 0;
        
        while (mcount > 0) {
            System.out.println("F".repeat(n));
            drinkedBottles += n;
            count++;
            mc = n + ost;
            System.out.println("E".repeat(mc));
            ost = mc % m;
            n = mc / m;
            count++;
            mc = ost;
            mcount = n;
        }
        
        System.out.println(drinkedBottles + " " + (count - 1));
    }
}

File: num2.kt
import java.util.Scanner
fun main() {
    val scanner = Scanner(System.`in`)
    printLn("full bottles: ")
    var n = 0
    n = scanner.nextInt()
    printLn("empty bottles for exchange: ")
    var m = 0
    m = scanner.nextInt()
    var mc = n
    var mcount = n
    var count = 0
    var ost = 0
    var drinkedBottles = 0
    
    while (mcount > 0) {
        println("F".repeat(n))
        drinkedBottles += n
        count++
        mc = n + ost
        println("E".repeat(mc))
        ost = mc % m
        n = mc / m
        count++
        mc = ost
        mcount = n
    }
    
    println("$drinkedBottles ${count-1}")
}
